modules/aws/src/smithy4s/aws/AwsCredentialsFile.scala

```diff
@@ -0,0 +1,158 @@
+/*
+ *  Copyright 2021-2022 Disney Streaming
+ *
+ *  Licensed under the Tomorrow Open Source Technology License, Version 1.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *     https://disneystreaming.github.io/TOST-1.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package smithy4s.aws
+
+import cats.effect.Resource
+import cats.effect.Sync
+import cats.syntax.all._
+import java.nio.file.Paths
+import smithy4s.aws.kernel.AWS_ACCESS_KEY_ID
+import smithy4s.aws.kernel.AWS_SECRET_ACCESS_KEY
+import smithy4s.aws.kernel.AWS_SESSION_TOKEN
+import smithy4s.aws.kernel.AwsCredentials
+
+final case class AwsCredentialsFile(
+    default: Option[AwsCredentials],
+    profiles: Map[String, AwsCredentials]
+)
+
+object AwsCredentialsFile {
+  def loadFromDisk[F[_]](home: String, profile: Option[String])(implicit
```

Comment:
don't put `home` there, but rather the full location of the file (using fs2.io.files.Path).

Reply:
Also, let's add a test for this with a temp file (created with fs2.io) to ensure that it works on the 3 platforms